# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/texture_cache.rs
# Commit: 0edb6b6f17b2
# Full Hash: 0edb6b6f17b24c737d053df19730e9c9e61e17ca
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r ee0de0eac18d -r 0edb6b6f17b2 gfx/wr/webrender/src/texture_cache.rs
--- a/gfx/wr/webrender/src/texture_cache.rs	Thu Dec 17 22:26:08 2020 +0000
+++ b/gfx/wr/webrender/src/texture_cache.rs	Thu Dec 17 22:26:10 2020 +0000
@@ -833,7 +833,6 @@
             uv_rect_handle,
             texture_id: TextureSource::TextureCache(
                 texture_id,
-                ImageBufferKind::Texture2D,
                 swizzle,
             ),
             uv_rect,
@@ -1149,7 +1148,6 @@
     pub fn alloc_render_target(
         &mut self,
         size: DeviceIntSize,
-        num_layers: usize,
         format: ImageFormat,
     ) -> CacheTextureId {
         let texture_id = self.next_id;
@@ -1157,12 +1155,12 @@
 
         // Push a command to allocate device storage of the right size / format.
         let info = TextureCacheAllocInfo {
-            target: ImageBufferKind::Texture2DArray,
+            target: ImageBufferKind::Texture2D,
             width: size.width,
             height: size.height,
             format,
             filter: TextureFilter::Linear,
-            layer_count: num_layers as i32,
+            layer_count: 1,
             is_shared_cache: false,
             has_depth: false,
         };